src/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -119,33 +119,44 @@ public function process(File $phpcsFile, $stackPtr)
             return;
         }
 
+        // Check there are no blank lines in the preamble for the property,
+        // but ignore blank lines _within_ attributes as that's not the concern of this sniff.
         if ($tokens[$commentEnd]['line'] !== ($tokens[$stackPtr]['line'] - 1)) {
             for ($i = ($commentEnd + 1); $i < $stackPtr; $i++) {
-                if ($tokens[$i]['column'] !== 1) {
+                // Skip over the contents of attributes.
+                if (isset($tokens[$i]['attribute_closer']) === true) {
+                    $i = $tokens[$i]['attribute_closer'];
                     continue;
                 }
 
-                if ($tokens[$i]['code'] === T_WHITESPACE
-                    && $tokens[$i]['line'] !== $tokens[($i + 1)]['line']
+                if ($tokens[$i]['column'] !== 1
+                    || $tokens[$i]['code'] !== T_WHITESPACE
```

Comment:
Have you considered adding a test where this condition is `true` and `$tokens[$i]['column']` is `false`? As far as I can check, in the current tests whenever `$tokens[$i]['code'] !== T_WHITESPACE` is `true`, `$tokens[$i]['column'] !== 1` is also `true`. Thus the former is never executed.

Reply:
No, I did not as those are pre-existing conditions which I've not changed. I've just grouped them into one `if - continue` as having multiple didn't make much sense.

Writing tests for pre-existing code is outside the scope of this PR.

Reply:
That is fair.

I asked because this condition was inverted. There are test cases that cover the first token of a line between the comment end and the function declaration being `T_WHITESPACE`, but there is no test case for it not being `T_WHITESPACE`.

Reply:
They are inverted because the flow has changed from "complies with condition -> throw error" to "does not comply with condition -> continue without error".

That is not a functional change. The behaviour of the sniff with regards to those conditions is unchanged - and outside the scope of this PR.

Reply:
I'm not saying the sniff couldn't use more tests, most sniffs can. Those additional tests though, do not belong with the _functional_ change this PR is making, so should be added in a separate PR.

Reply:
Sounds good to me.